modules/custom_terraform/custom_terraform.py

```diff
@@ -14,12 +15,26 @@ def __init__(self, module: "Module", layer: "Layer"):
 
     def process(self, module_idx: int) -> None:
         current_desc = self.module.desc
-        self.module.module_dir_path = os.path.join(
-            os.path.abspath(os.path.dirname(self.layer.path)),
-            self.module.data["path_to_module"].strip("./"),
-        )
+        path_to_module: str = self.module.data["path_to_module"]
+        path_to_layer: str = os.path.abspath(os.path.dirname(self.layer.path))
+        module_version: str = self.module.data.get("version", None) or ""
+        if path_to_module.startswith("./"):
+            path_to_module = path_to_module.strip("./")
+            self.module.module_dir_path = os.path.join(path_to_layer, path_to_module)
+        elif path_to_module.startswith("../"):
+            self.module.module_dir_path = os.path.join(path_to_layer, path_to_module)
+        else:
+            # If this is the case, then this refers to a remote module as per
+            # https://www.terraform.io/language/modules/sources
+            self.module.module_dir_path = path_to_module
+            if module_version == "":
+                raise UserErrors(
+                    "module_version must be given to custom terraform for non-local modules"
```

Comment:
do we need to handle this, would terraform handle this logic?
for example, a github repo would not beed a module version but we would throw this error.

Reply:
I didn't know that, but this is a really nice error message

Reply:
here is the use case with github I was trying to explain earlier
```yaml
  - type: custom-terraform
    name: bucket
    source: "github.com/hashicorp/example
    # version not specified
```
https://www.terraform.io/language/modules/sources#github

but this code would throw the error `module_version must be given...`

trying to implement this exception handling seems to be the responsibility of terraform here, right?